### PR TITLE
First commit for 404 internal links handler

### PR DESCRIPTION
**Pull request by nitinja**
### Broken Internal Links handling

**Test cases coming soon.**

This PR introduces a 404 links handler from internal sources - same domain.
It handles `navigate` and `404` events and sums views from same source/target combos that has both of above events.

## Related Issues
https://jira.corp.adobe.com/browse/SITES-27242

#### Sample data input
```
[
  {
    "id": "56ou",
    "host": "rum.hlx.page",
    "time": "2024-11-23T23:00:01.858Z",
    "timeSlot": "2024-11-23T23:00:00.000Z",
    "url": "https://www.petplace.com/a01",
    "userAgent": "mobile:android:blink",
    "weight": 200,
    "events": [
      {
        "checkpoint": "navigate",
        "target": "visible",
        "source": "https://www.petplace.com/a01nf",
        "timeDelta": 1858
      },
      {
        "checkpoint": "navigate",
        "target": "visible",
        "source": "https://www.petplace.com/a01nf",
        "timeDelta": 1654
      },
      {
        "checkpoint": "404",
        "target": "",
        "source": "https://www.petplace.com/a01nf",
        "timeDelta": 218
      }
    ]
  },
  {
    "id": "2no",
    "host": "rum.hlx.page",
    "time": "2024-11-22T18:00:00.544Z",
    "timeSlot": "2024-11-22T18:00:00.000Z",
    "url": "https://www.petplace.com/a01",
    "userAgent": "desktop:windows:blink",
    "weight": 1800,
    "events": [
      {
        "checkpoint": "navigate",
        "target": "visible",
        "source": "https://www.petplace.com/a02nf",
        "timeDelta": 542
      },
      {
        "checkpoint": "404",
        "target": "xxxx",
        "source": "https://www.petplace.com/a02nf",
        "timeDelta": 45
      }
    ]
  },
  {
    "id": "4FGLNSev",
    "host": "rum.hlx.page",
    "time": "2024-11-23T23:00:33.919Z",
    "timeSlot": "2024-11-23T23:00:00.000Z",
    "url": "https://www.petplace.com/a01",
    "userAgent": "desktop:mac:blink",
    "weight": 100,
    "events": [
      {
        "checkpoint": "navigate",
        "target": "hidden",
        "source": "https://www.petplace.com/a03",
        "timeDelta": -3599620
      }
    ]
  },
  {
    "id": "Hdq",
    "host": "rum.hlx.page",
    "time": "2024-11-24T00:00:00.111Z",
    "timeSlot": "2024-11-24T00:00:00.000Z",
    "url": "https://www.petplace.com/a01",
    "userAgent": "mobile:android:blink",
    "weight": 100,
    "events": []
  },
  {
    "id": "9DTknv",
    "host": "rum.hlx.page",
    "time": "2024-11-24T00:00:14.264Z",
    "timeSlot": "2024-11-24T00:00:00.000Z",
    "url": "https://www.petplace.com/a01",
    "userAgent": "mobile:android:blink",
    "weight": 100,
    "events": []
  },
  {
    "id": "9LOQXfu",
    "host": "rum.hlx.page",
    "time": "2024-11-22T23:00:00.581Z",
    "timeSlot": "2024-11-22T23:00:00.000Z",
    "url": "https://www.petplace.com/a01",
    "userAgent": "mobile:ios:webkit",
    "weight": 100,
    "events": [
      {
        "checkpoint": "404",
        "target": "https://www.petplace.com/a04nf",
        "source": "",
        "timeDelta": 198
      }
    ]
  },
  {
    "id": "047B",
    "host": "rum.hlx.page",
    "time": "2024-11-20T20:00:01.683Z",
    "timeSlot": "2024-11-20T20:00:00.000Z",
    "url": "https://www.petplace.com/ax02",
    "userAgent": "desktop:windows",
    "weight": 700,
    "events": [
      {
        "checkpoint": "404",
        "target": "ddd",
        "source": "https://www.petplace.com/ax02nf",
        "timeDelta": 259
      },
      {
        "checkpoint": "navigate",
        "target": "hidden",
        "source": "https://www.petplace.com/ax02nf",
        "timeDelta": 1680
      }
    ]
  },
  {
    "id": "048C",
    "host": "rum.hlx.page",
    "time": "2024-11-20T20:00:01.683Z",
    "timeSlot": "2024-11-20T20:00:00.000Z",
    "url": "https://www.petplace.com/ax02",
    "userAgent": "desktop:windows",
    "weight": 500,
    "events": [
      {
        "checkpoint": "404",
        "target": "ddd",
        "source": "https://www.petplace.com/ax02nf",
        "timeDelta": 259
      },
      {
        "checkpoint": "navigate",
        "target": "hidden",
        "source": "https://www.petplace.com/ax02nf",
        "timeDelta": 1680
      }
    ]
  }
]

```

#### Sample output
```
[
      {
        "views": 1800,
        "url_to": "https://www.petplace.com/a01",
        "url_from": "https://www.petplace.com/a02nf",
      },
      {
        "views": 1200,
        "url_to": "https://www.petplace.com/ax02",
        "url_from": "https://www.petplace.com/ax02nf",
      },
      {
        "views": 200,
        "url_to": "https://www.petplace.com/a01",
        "url_from": "https://www.petplace.com/a01nf",
      }
    ]
```

Please ensure your pull request adheres to the following guidelines:
- [x] make sure to link the related issues in this description
- [x] when merging / squashing, make sure the fixed issue references are visible in the commits, for easy compilation of

### CODE REVIEW

**Review by ekremney — APPROVED** [review]

File: packages/spacecat-shared-rum-api-client/src/functions/404-internal-links.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+      acc[urlCombination].views = facet.metrics.views.sum;
+      const [urlTo, urlFrom] = facet.value.split(DELIMITER);
+      acc[urlCombination].url_to = urlTo;
+      acc[urlCombination].url_from = urlFrom;
+
+      return acc;
+    }, {});
+
+  return Object.values(brokenInternalLinksViews);
```
Comment (ekremney) [inline packages/spacecat-shared-rum-api-client/src/functions/404-internal-links.js:62]: if I understand the logic correctly, no need to `reduce` + `Object.values` here; a simple `map` would suffice

```suggestion
  return dataChunks.facets.uniqueUrlCombinations.map((facet) => {
      const [url_to, urlFrom] = facet.value.split(DELIMITER);
      
      return {
         views: facet.metrics.views.sum,
         url_to,
         url_from,
      }
    });
```

**Review by ekremney — APPROVED** [review]
please add the functional tests as well

File: packages/spacecat-shared-rum-api-client/src/functions/404-internal-links.js (hunk 1 above)
Comment (nitinja) [reply packages/spacecat-shared-rum-api-client/src/functions/404-internal-links.js:62]: good observation, we are not really aggregating anything here, so map will be good enough here.

**Comment by nitinja** [comment]
- All tests are passing
- I have done some minor updates, but core code remains intact
- Result data verified

**Pull request merged**